src/Icons/doc/index.rst

```diff
@@ -90,20 +158,127 @@ Usage
+added to all icons unless overridden by the second argument of the ``ux_icon`` function.
+
+.. code-block:: yaml
+
+    # config/packages/ux_icons.yaml
+    ux_icons:
+        default_icon_attributes:
+            fill: currentColor
+
+Now, all icons will have the ``fill`` attribute set to ``currentColor`` by default.
+
+.. code-block:: twig
+
+    # renders "user-profile.svg" with fill="currentColor"
+    {{ ux_icon('user-profile') }}
+
+    # renders "user-profile.svg" with fill="red"
+    {{ ux_icon('user-profile', {fill: 'red'}) }}
+
+Accessibility
+-------------
 
-    <twig:UX:Icon name="user-profile" class="w-4 h-4" /> <!-- renders "user-profile.svg" -->
+Icons are a great way to add visual elements to your website, but they can be a challenge
+for accessibility.
 
-    <twig:UX:Icon name="sub-dir:user-profile" class="w-4 h-4" /> <!-- renders "sub-dir/user-profile.svg" (sub-directory) -->
+According to the `W3C <https://design-system.w3.org/styles/svg-icons.html>`_, there are
+three methods to improve icons accessibility, depending on the context:
 
-    <twig:UX:Icon name="flowbite:user-solid" /> <!-- renders "flowbite:user-solid" from ux.symfony.com -->
+    * **Decorative**: Icons that are purely decorative and do not convey any meaning or function.
+    * **Informative**: Icons that convey information or a function.
+    * **Functional**: Icons that are interactive and perform a function.
 
-Caching
--------
+Decorative Icons
+~~~~~~~~~~~~~~~~
+
+If the icon is purely decorative and does not convey any meaning or function, it should be
+hidden from screen readers using the ``aria-hidden`` attribute.
+
+.. code-block:: html+twig
+
+    <a href="/profile">
+        {{ ux_icon('user-profile', {class: 'w-4 h-4', 'aria-hidden': true}) }}
+        Back to profile
+    </a>
+
+Informative Icons
+~~~~~~~~~~~~~~~~~
+
+If the icon conveys information or a function, it should be given a text alternative that presents
+the same content or function, so that it can fulfill the same purpose.
+
+.. code-block:: html+twig
+
+    Today's weather:
+    {{ ux_icon('cloud-rain', {'aria-label': 'Rainy weather'}) }}
+
+Functional Icons
+~~~~~~~~~~~~~~~~
+
+If the icon is interactive and performs a function, it should be given a text alternative that
+presents the same content or function, so that it can fulfill the same purpose.
+
+.. code-block:: html+twig
+
+    {{ ux_icon('user-profile', {class: 'w-4 h-4', 'aria-label': 'User Profile'}) }}
+
+.. note::
+
+    The ``aria-label`` attribute is used to provide a label for the icon. It is read by
+    screen readers and other assistive technologies.
+
+.. note::
+
+    The ``aria-hidden`` attribute is used to hide the icon from screen readers and other
+    assistive technologies.
+
+
+
+Performances
+------------
+
+The UXIcon component is designed to be as fast as possible, while offering a
+great deal of flexibility. The following are some of the optimizations made to
```

Comment:
I'd remove the mention to "flexibility". Let's focus on saying that UX Icons is fast and how is that possible.